src/server/auth/server/testing/auth_test.go

```diff
@@ -2346,17 +2346,46 @@ func TestDeleteAll(t *testing.T) {
 func TestCreateProject(t *testing.T) {
 	t.Parallel()
 	client := envWithAuth(t).PachClient
+	rootClient := tu.AuthenticateClient(t, client, auth.RootUser)
 	alice := tu.Robot(tu.UniqueString("alice"))
 	aliceClient := tu.AuthenticateClient(t, client, alice)
+	bob := tu.Robot(tu.UniqueString("bob"))
+	bobClient := tu.AuthenticateClient(t, client, bob)
 
 	// create a project and check the caller is the owner
 	projectName := tu.UniqueString("project" + t.Name())
 	require.NoError(t, aliceClient.CreateProject(projectName))
 	require.Equal(t, tu.BuildBindings(alice, auth.ProjectOwner), tu.GetProjectRoleBinding(t, aliceClient, projectName))
 
+	// verify that bob cannot modify role binding for given project
+	_, err := bobClient.ModifyRoleBinding(bobClient.Ctx(), &auth.ModifyRoleBindingRequest{
+		Principal: bob,
+		Roles:     []string{auth.ProjectOwner},
+		Resource:  &auth.Resource{Type: auth.ResourceType_PROJECT, Name: projectName},
+	})
+	require.YesError(t, err)
+	require.Matches(t, `.+not authorized to perform this operation - needs permissions \[PROJECT_MODIFY_BINDINGS\] on PROJECT.+`, err.Error())
+
+	// Alice can give Bob project owner access
+	_, err = aliceClient.ModifyRoleBinding(aliceClient.Ctx(), &auth.ModifyRoleBindingRequest{
+		Principal: bob,
+		Roles:     []string{auth.ProjectOwner},
+		Resource:  &auth.Resource{Type: auth.ResourceType_PROJECT, Name: projectName},
+	})
+	require.NoError(t, err)
+
+	// Bob can now remove Alice as project owner
+	_, err = bobClient.ModifyRoleBinding(bobClient.Ctx(), &auth.ModifyRoleBindingRequest{
+		Principal: alice,
+		Roles:     []string{},
+		Resource:  &auth.Resource{Type: auth.ResourceType_PROJECT, Name: projectName},
+	})
+	require.NoError(t, err)
+
```

Comment:
Any tests that auth.ProjectOwner can't be applied to repos and things like that?  Though it does seem like we don't do these checks for other things, perhaps mistakenly.  (You can `pachctl auth set enterprise repoOwner user:kilgore@trout` without getting an error.)

Reply:
Good question. We don't have code that prevents binding roles to resource types that are incompatible with the role's `auth.Role.ResourceTypes`. Currently, this is just used to tell the user, which roles allows the user to access a resource type.